utils/zfp.c

```diff
@@ -531,8 +553,16 @@ int main(int argc, char* argv[])
         fprintf(stderr, "cannot create compressed file\n");
         return EXIT_FAILURE;
       }
+      /* TODO: think of a clean way to check for offset header, possibly execution policy */
+	    if (chunk_size) {
+        /* write the offset header to the file */
+        if (fwrite(offset_table, sizeof(unsigned long long), chunks, file) != chunks) {
```

Comment:
I think you need to align with @lindstro and @mclarsen on how to handle writing the offset table. Few questions that pop up:
- Should it go to the header or a separate file?
- If it goes to the header, do we version the header to preserve backwards-compatibility?

Reply:
This will break compatibility in the sense that existing code will not be able to read a file that stores these offsets.  We are planning on making several changes to the compressed format for our September release, and that would be the right time to address this.  I'm afraid we can't officially adopt changes like this until then.

Perhaps we can make an experimental branch for evaluating parallel decompression and experimenting with different ways of encoding the offset table.  For instance, I had a discussion with @LennartNoordsij about alternative solutions that are not based on storing offsets.

Reply:
An experimental branch seems like the best way to go forward for now. I will also add my CUDA decompression implementation which is based on the same offset table implementation as this OpenMP version. I will combine them both in a branch "Parallel decompression".